pyiceberg/expressions/parser.py

```diff
@@ -233,10 +233,14 @@ def handle_not(result: ParseResults) -> Not:
 
 
 def handle_and(result: ParseResults) -> And:
+    if len(result[0]) > 2:
+        return And(result[0][0], result[0][1], *result[0][2:])
     return And(result[0][0], result[0][1])
```

Comment:
What do you think of:
```suggestion
    return And(*result[0])
```

Reply:
Haha just came to the realization I can do this and don't need to differentiate based on the length....pushed up the change. Way better